### DISCUSSION TITLE
Is there a way to get file dependencies for a filepath without compiling ?

### DISCUSSION THREAD

**Discussion by Lumengine** [shader-slang/slang#2690] [Q&A] [answered]
Hi,
I'm doing some shaders hot reload and I was wondering if by any chance we can get file dependencies for a .slang without compiling it ?  I saw that there is some stuff in ICompileRequest but I'm not using compileRequest at all. I would love some tips if it's possible.
Thank you ! 
Baptiste

**Comment by tangent-vector** [comment, marked as answer]
There is a `NO_CODEGEN` flag that can be used from the C API. Using that flag should eliminate some amount of the compilation time (but not all of it), and might be sufficient for your needs.

**Reply by tangent-vector** [reply]
The same basic question has come up on another front recently, so it is possible that there is more we could be doing in the compiler to make dependency resolution into a discrete step before other semantic checks are performed.

**Comment by Lumengine** [comment]
Thank you for your answer @tangent-vector  ! Another questions: Is there a way to know if a member of a struct is really used in the final shader. For example defining a RaytracingAccelerationStructure in a struct but for some shaders that import this struct, it won't use the AS. The reflection always give the resource right ? I also tried to do an extension of this struct and define this member in the extension but it seems to not be allowed ? You can only defined new functions in extension ? Thank you

**Reply by Lumengine** [reply]
OK I saw in the documentation that it is supposed to work, defining new member in extension. I will look at my code why it's not working.

**Reply by Lumengine** [reply]
For example :
```

__exported import Camera.Camera;
struct Scene
{
    Camera camera;
}

/* Scene extension with RT AS*/
extension Scene
{
    RaytracingAccelerationStructure rtAccelerationStructure;
};


struct PrimaryRayPayload
{
    float3 color;
    float  depth;
};

ParameterBlock<Scene> scene;

[shader("raygeneration")]
void rayGen()
{
	uint2 launchIndex = DispatchRaysIndex().xy;
    uint2 launchDim = DispatchRaysDimensions().xy;

    PrimaryRayPayload payload = {float3(0, 0, 0), 0.0};

    Ray ray = scene.camera.ComputeRayPinhole(launchIndex, launchDim);

    TraceRay(scene.rtAccelerationStructure,             // Acceleration structure
             RAY_FLAG_NONE,
             ~0,                                        // Instance inclusion mask - all instances are visible
             0,                                         // Ray contribution to hit group index (aka ray type)
             0,                                         // Multiplier for geometry contribution to hit
             0,                                         // Miss shader index
             ray.ToRayDesc(),
             payload);
}
```
Generate GLSL and HLSL files but that missed the rtAccelerationStructure in glsl file so glslang throw an error. Any idea ?
